src/main/java/seedu/address/logic/parser/ListCommandParser.java

```diff
@@ -0,0 +1,68 @@
+package seedu.address.logic.parser;
+
+import seedu.address.logic.commands.AddCommand;
+import seedu.address.logic.commands.ListCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import static seedu.address.logic.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.*;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ALIAS;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+public class ListCommandParser implements Parser<ListCommand>{
+    @Override
+    public ListCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ALIAS);
+
+
+        List<Predicate<Person>> predicates = new ArrayList<>();
+
+        if (!hasAtLeastOnePrefixPresent(argMultimap, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ALIAS)
+                || !argMultimap.getPreamble().isEmpty()) {
```

Comment:
Could this boolean condition be shortened instead to just a single boolean variable that is more descriptive on what this condition represents?

Reply:
This was actually from the original AB3 code. I agree with you, it can be shortened to a single variable

Reply:
Alright. Are you planning to fix this in this PR or in a later PR?

Reply:
Fixed